David/feat/signup

Ringayend opened the pull request.

Ringayend committed:
SignUp and NavBar

Design the route behaviour and create a sign up form
Navbar and Signup form

Add visual Navbar and Signup form
Register container

Reducer, Container, actions, constants was created
Modification of register container

action, constants and reducer modification
Final register feature

fix the form issue

Ringayend committed:
Merge branch 'master' into david/feat/signup

TheoGauchoux commented:
You don't manage the errors emitted by the API. You must display a hint to indicate the error on the form:
* Password is too short (<8 characters) or empty
* Email already taken or empty
* Fullname empty

Also, when the user is correctly registered, indicate it with a message, then redirect him to the main page (for the moment, later it will be the login page)

TheoGauchoux committed:
Build fix

TheoGauchoux commented on src/components/navbar/Navbar.jsx:
```diff
@@ -0,0 +1,37 @@
+  )
+}
+
+const { Header } = Layout
```
Put it to the top after imports

TheoGauchoux commented on src/components/navbar/Navbar.jsx:
```diff
@@ -0,0 +1,37 @@
+import routes from '../../config/routes'
+import './style.css'
+
+const renderLinks = () => {
```
Use the Class syntax like containers or complexe components. Put this method inside of it and don't forget to `.bind` it.

TheoGauchoux commented on src/components/navbar/Navbar.jsx:
```diff
@@ -0,0 +1,37 @@
+      theme="dark"
+      mode="horizontal"
+      defaultSelectedKeys={['Home']}
+      style={{ lineHeight: '64px' }}
```
The style needs to be in a specific `.css` file

TheoGauchoux commented on src/components/register/Register.jsx:
```diff
@@ -0,0 +1,84 @@
+      }
+    })
+  }
+  render() {
```
Must be a space before the method

TheoGauchoux commented on src/components/register/Register.jsx:
```diff
@@ -0,0 +1,84 @@
+    })
+  }
+  render() {
+    const { Content } = Layout
```
Put it to the top of the field, with `const FormItem = Form.Item`

TheoGauchoux commented on src/components/register/Register.jsx:
```diff
@@ -0,0 +1,84 @@
+            </FormItem>
+            <FormItem className="signupFormItemFooter">
+              <Button type="primary" htmlType="submit" className="signup-form-button">
+            Sign up
```
Also for `Or`, issue with the indents

TheoGauchoux commented on src/config/routes.jsx:
```diff
@@ -0,0 +1,5 @@
+export default {
+  root: '/',
+
```
Useless space

TheoGauchoux commented on src/containers/register/reducer.js:
```diff
@@ -0,0 +1,25 @@
+import { REGISTER } from './constants'
+
+const initialState = {
```
Use an objet, not an array. You will need to handle errors coming from the API, I suggest you to use the state's shape:
```javascript
{
  success: false,
  error: '',
}
```

For `_SUCCESS` action update `success` to `true` (and so show the info message in the component). For `_ERROR`, put the error into `error` and display it in the component.

TheoGauchoux commented on src/reducers.js:
```diff
@@ -4,11 +4,13 @@ import boardsReducer from './containers/boards/reducer'
   currentBoard: boardReducer,
   lists: listsReducer,
   cards: cardsReducer,
+  register: registerReducer,
```
Maybe `registering` is a better word to describe this sub-state ?

TheoGauchoux reviewed (changes requested):
Errors handling and success message missing

TheoGauchoux commented:
Also, the form is not aligned horizontally and a whitespace is missing after the `SignUp` button:

<img width="1680" alt="screen shot 2017-10-25 at 20 55 54" src="https://user-images.githubusercontent.com/6637214/32017419-ecde66c8-b9c6-11e7-9611-6cb1fb35e23f.png">

Ringayend committed:
Improvements  register 

First improvements for the register feature such as formatting the code and create class for the navbar.

Ringayend committed:
Merge remote-tracking branch 'origin/david/feat/signup' into david/feat/signup

Ringayend committed:
RegExp and Style

Add the regular expression for the password and modify the style to be responsive.

TheoGauchoux committed:
Merge branch 'master' into david/feat/signup

TheoGauchoux commented on src/components/register/Register.jsx:
```diff
@@ -0,0 +1,95 @@
+                  </FormItem>
+                  <FormItem className="signupFormItemFooter">
+                    <Button type="primary" htmlType="submit" className="signup-form-button"> Sign up </Button>
+                    <div className="signupLogin">Or <a href="/">Login</a></div>
```
Instead of `<a>` prefer the use of `<Link>` provided by the `react-router-dom` package. You can find an exemple in the `BoardPreview.jsx` file

Ringayend committed:
Registration confirmed

Add a modal to confirm the registration.

Ringayend committed:
Merge branch 'master' into david/feat/signup

Ringayend committed:
Redirection after register

the user is automatically redirected after the registration process.

Ringayend committed:
Merge remote-tracking branch 'origin/david/feat/signup' into david/feat/signup

Ringayend committed:
Add improvements

Link tag instead of  a tag
Corrrect password synthese

Ringayend committed:
Fix link issue

Link coming from react-router-dom
